media-libs/imlib2/imlib2-1.5.1.ebuild

```diff
@@ -0,0 +1,96 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit multilib-minimal toolchain-funcs
+
+DESCRIPTION="Version 2 of an advanced replacement library for libraries like libXpm"
+HOMEPAGE="https://www.enlightenment.org/"
+SRC_URI="https://downloads.sourceforge.net/enlightenment/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="bzip2 cpu_flags_x86_mmx cpu_flags_x86_sse2 doc gif jpeg mp3 nls png static-libs tiff X zlib"
+
+RDEPEND="
+	media-libs/freetype:2[${MULTILIB_USEDEP}]
+	bzip2? ( app-arch/bzip2[${MULTILIB_USEDEP}] )
+	gif? ( media-libs/giflib[${MULTILIB_USEDEP}] )
+	jpeg? ( ~virtual/jpeg-0:0=[${MULTILIB_USEDEP}] )
+	mp3? ( media-libs/libid3tag[${MULTILIB_USEDEP}] )
+	nls? ( sys-devel/gettext )
+	png? ( >=media-libs/libpng-1.6.10:0=[${MULTILIB_USEDEP}] )
+	tiff? ( >=media-libs/tiff-4.0.3-r6:0[${MULTILIB_USEDEP}] )
+	X? (
+		x11-libs/libX11[${MULTILIB_USEDEP}]
+		x11-libs/libXext[${MULTILIB_USEDEP}]
+	)
+	zlib? ( sys-libs/zlib[${MULTILIB_USEDEP}] )
+"
+DEPEND="
+	${RDEPEND}
+	virtual/pkgconfig[${MULTILIB_USEDEP}]
+	doc? ( app-doc/doxygen )
+	X? (
+		x11-proto/xextproto[${MULTILIB_USEDEP}]
+		x11-proto/xproto[${MULTILIB_USEDEP}]
+	)
+"
+
+src_prepare() {
+	default
+
+	chmod a+rx gendoc || die
```

Comment:
Isn't it enough to make it `+x`? Provided that it's needed at all.

Reply:
>-rwxr-xr-x 1 joonas joonas    390 Mar 16  2013 gendoc

Seems to be executable out of the box. However it does nothing and doesnt even work, there are already docs in HTML format under doc/. I removed doc USE flag altogether in favor of HTML_DOCS.
```diff
-IUSE="bzip2 cpu_flags_x86_mmx cpu_flags_x86_sse2 doc gif jpeg mp3 nls png static-libs tiff X zlib"
+IUSE="bzip2 cpu_flags_x86_mmx cpu_flags_x86_sse2 gif jpeg mp3 nls png static-libs tiff X zlib"

 DEPEND="
 	${RDEPEND}
 	virtual/pkgconfig[${MULTILIB_USEDEP}]
-	doc? ( app-doc/doxygen )
 	X? (
 		x11-proto/xextproto[${MULTILIB_USEDEP}]
 		x11-proto/xproto[${MULTILIB_USEDEP}]
 	)


-src_prepare() {
-	default
-
-	chmod a+rx gendoc || die
-}
-

-multilib_src_compile() {
-	V=1 emake || die
-
-	if use doc ; then
-		if [[ -x ./gendoc ]] ; then
-			./gendoc || die
-
-		elif emake -j1 -n doc >&/dev/null ; then
-			V=1 emake doc || die
-
-		fi
-	fi
-}

 multilib_src_install() {
+	HTML_DOCS=( doc/. )
 	einstalldocs
 
-	use doc && [[ -d doc ]] && local HTML_DOCS=( doc/. )
}
```